jetty-server/src/main/java/org/eclipse/jetty/server/Authentication.java

```diff
@@ -50,7 +50,9 @@ public Failed(String message)
         String getAuthMethod();
         UserIdentity getUserIdentity(); 
         boolean isUserInRole(UserIdentity.Scope scope,String role);
+        @Deprecated
         void logout();
+        Authentication logout(ServletRequest request);
```

Comment:
I think you can add another interface `Logout` that just has the logout method.  Then both `User` and `Deferred` can extend `Logout` as multiple inheritance is allows with interfaces.    I'm not sure if `Logout` should extend `Authentication` or not?

Reply:
Done. Is an Authentication.